kernel/src/process/program_loader/elf/load_elf.rs

```diff
@@ -507,26 +507,38 @@ pub fn init_aux_vec(
 }
 
 /// Maps the VDSO VMO to the corresponding virtual memory address.
+#[cfg(any(target_arch = "x86_64", target_arch = "riscv64"))]
 fn map_vdso_to_vm(process_vm: &ProcessVm) -> Option<Vaddr> {
+    use crate::vdso::{vdso_vmo, VdsoVmoLayout};
+
     let process_vmar = process_vm.lock_root_vmar();
     let root_vmar = process_vmar.unwrap();
     let vdso_vmo = vdso_vmo()?;
 
     let options = root_vmar
-        .new_map(VDSO_VMO_SIZE, VmPerms::empty())
+        .new_map(VdsoVmoLayout::SIZE, VmPerms::empty())
         .unwrap()
         .vmo(vdso_vmo.dup().unwrap());
 
-    let vdso_data_base = options.build().unwrap();
-    let vdso_text_base = vdso_data_base + 0x4000;
+    let vdso_vmo_base = options.build().unwrap();
+    let vdso_data_base = vdso_vmo_base + VdsoVmoLayout::DATA_OFFSET;
+    let vdso_text_base = vdso_vmo_base + VdsoVmoLayout::TEXT_OFFSET;
 
     let data_perms = VmPerms::READ | VmPerms::WRITE;
```

Comment:
This is also incorrect. Why should we allow users to write _shared_ vDSO data? It can break all the programs.
```
7f55a7689000-7f55a768d000 r--p 00000000 00:00 0                          [vvar]
7f55a768d000-7f55a768f000 r--p 00000000 00:00 0                          [vvar_vclock]
7f55a768f000-7f55a7691000 r-xp 00000000 00:00 0                          [vdso]
```

This isn't a problem introduced by this PR, so you don't have to fix it. However, if the fix is as simple as removing `VmPerms::WRITE`, you could create another commit to fix this issue.